### ISSUE TITLE
Catch urinary operator after keyword await as error

### ISSUE THREAD

**Issue by timnew** [eslint/eslint#4337]
**When does this rule warn? Please describe and show example code:**

To invert the boolean value returned from promise with `await`, we should put the `!` before the `await` keyword. But this is counterintuitive and error-prone.

e.g.
```js
function returnFalse() {
  return Promise.resolve(false);
}

async function test() {
  if(await !returnFalse()) {
    console.log('ok');
  }  
}

test();
```

We might expect `ok` to be displayed, but it never does. Since we apply `!` on the `promise` return by async function, which is meaningless in most cases. So it would be nice if any code that apply `!` after `await` rather than before.

Here is the correct `test()`:
```js
async function test() {
  if(! await returnFalse()) {
    console.log('ok');
  }  
}

```

**Is this rule preventing an error or is it stylistic?**

Yes it preventing error.

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

I think it will be a general issue when writing code with `async/await`. And it is kind of hard to find if you haven't aware it clearly.

**Are you willing to create the rule yourself?**

I think I can have a try

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by timnew** [comment]
This case should apply to all urinary operators, including `!`, `~`, and `-`.

**Comment by jimbolla** [comment]
I think you mean unary, not urinary. ;)

**Comment by ilyavolodin** [comment]
ESLint doesn't naively support experimental features (await, async). So you will need to create a plugin for this yourself.
